docs/developer-guide.md

```diff
@@ -43,6 +43,20 @@ You can undeploy Katib v1beta1 manifests from a Kubernetes cluster as follows:
 make undeploy
 ```
 
+## Technical and style guide
+
+The following guidelines apply primarily to Katib, 
+but other projects like training-operator might also adhere to them. 
+
+When coding:
+
+- Follow [effective go](https://go.dev/doc/effective_go) guidelines.
+
+Testing:
+
+- Use `cmp.Diff` instead of `reflect.Equal`, to provide useful comparisons.
+- Define test cases as maps instead of slices to avoid dependencies on te running order.
```

Comment:
Maybe we should say that:
```suggestion
- Define test cases as maps instead of slices to avoid dependencies on the running order. Set the test case name in the map key.
```

Reply:
That’s a great clarification!

Reply:
Done.